app/tests/integration/nimbus/test_remote_settings.py

```diff
@@ -90,3 +94,49 @@ def test_end_experiment_and_reject_end(
     Dashboard(selenium, kinto_review_url).open().reject()
 
     SummaryPage(selenium, experiment_url).open().wait_for_rejected_alert()
+
+
+@pytest.mark.run_once
+def test_create_new_experiment_approve_check_targeting(
+    selenium,
+    base_url,
+    kinto_url,
+    kinto_review_url,
+    experiment_url,
+    default_data,
+    create_experiment,
+    slugify,
+):
+    default_data.audience.targeting = (
+        BaseExperimentAudienceTargetingOptions.TARGETING_MAC_ONLY
+    )
+    default_data.audience.locale = "English (Canadian)"
+    default_data.audience.countries = "Canada"
+    create_experiment(selenium).launch_and_approve()
+
+    Login(selenium, kinto_url).open().login()
+    Dashboard(selenium, kinto_review_url).open().approve()
+    SummaryPage(selenium, experiment_url).open().wait_for_live_status()
+    title = default_data.public_name
+    base_url = urlparse(base_url)
+    json_url = f"https://{base_url.netloc}/api/v6/experiments/{slugify(title)}"
+    # Get experiment JSON and parse
+    experiment_json = requests.get(f"{json_url}", verify=False).json()
+    targetting = experiment_json["targeting"]
+    expression = []
+    environment = []
+    for item in targetting.split("&&"):
+        if "locale" in item or "region" in item:
+            environment.append(item)
+        else:
+            expression.append(item)
+    if environment is False:
+        environment = "en-US"
+    expression = "&&".join(expression)
+    environment = "&&".join(environment)
+    # Inject filter expression
+    selenium.get("about:blank")
+    with open("nimbus/utils/filter_expression.js") as js:
+        with selenium.context(selenium.CONTEXT_CHROME):
+            script = selenium.execute_script(js.read(), 90.1, environment, expression)
+    assert script is not None
```

Comment:
It might be nice to create a Page object that wraps this so we can interact with it at a higher level in the test body?